File: tests/latest/vbnet/code/src/Test.Unit.Library/HelloWorldTest.vb
Imports System.Text
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Reflection
Imports NBuildKit.Test.VbNet.Library
Imports System.Globalization

<TestClass()> Public Class HelloWorldTest

    Private Function AssemblyName(a As Assembly) As String
        Dim attr As AssemblyTitleAttribute = TryCast(a.GetCustomAttribute(GetType(AssemblyTitleAttribute)), AssemblyTitleAttribute)
        Return attr.Title
    End Function

    Private Function AssemblyVersion(a As Assembly) As String
        Dim attr As AssemblyInformationalVersionAttribute = TryCast(a.GetCustomAttribute(GetType(AssemblyInformationalVersionAttribute)), AssemblyInformationalVersionAttribute)
        Return attr.InformationalVersion
    End Function

    <TestMethod()> Public Sub SayHello()
        Dim helloWorld As HelloWorld = New HelloWorld()
        Dim txt As String = helloWorld.SayHello()
        Dim expected As String = String.Format( _
            CultureInfo.InvariantCulture, _
            "Hello world from: {0} [{1}]", _
            AssemblyName(GetType(HelloWorld).Assembly), _
            AssemblyVersion(GetType(HelloWorld).Assembly))
        Assert.AreEqual(expected, txt)
    End Sub

End Class

File: tests/oldest/csharp/src/Test.Unit.Library/HelloWorldTest.cs
using NBuildKit.Test.CSharp.Library;
using NUnit.Framework;
using System;
using System.Globalization;
using System.Reflection;

namespace Test.Unit.Library
{
    [TestFixture]
    public class HelloWorldTest
    {
        private string AssemblyName(Assembly assembly)
        {
            var attribute = (AssemblyTitleAttribute)assembly.GetCustomAttribute(typeof(AssemblyTitleAttribute));
            return attribute.Title;
        }

        private string AssemblyVersion(Assembly assembly)
        {
            var attribute = (AssemblyInformationalVersionAttribute)assembly.GetCustomAttribute(typeof(AssemblyInformationalVersionAttribute));
            return attribute.InformationalVersion;
        }

        [Test]
        public void SayHello()
        {
            var helloWorld = new HelloWorld();
            var text = helloWorld.SayHello();
            var expected = string.Format(
                CultureInfo.InvariantCulture,
                "Hello world from: {0} [{1}]",
                AssemblyName(typeof(HelloWorld).Assembly),
                AssemblyVersion(typeof(HelloWorld).Assembly));
            Assert.AreEqual(expected, text);
        }
    }
}
